spec/lib/administrate/search_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "spec_helper"
+require "support/constant_helpers"
+require "administrate/fields/string"
+require "administrate/fields/email"
+require "administrate/fields/number"
+require "administrate/search"
+
+class MockDashboard
+  ATTRIBUTE_TYPES = {
+    name: Administrate::Field::String,
+    email: Administrate::Field::Email,
+    phone: Administrate::Field::Number,
+  }
+end
+
+describe Administrate::Search do
+  describe "#run" do
+    let(:resolver) do
```

Comment:
Absolutely, feel free to point out anywhere that you'd prefer changed to match your style. I'll get the hang of it.